### PR TITLE
Upgrade for Wagtail 4.0

### PR DESCRIPTION
**Pull request by nickmoreton**
This PR build off of https://github.com/torchbox/wagtail-ab-testing/pull/48 which includes a Wagtail 3 upgrade.

Lets merge that before this one.

### CODE REVIEW

File: wagtail_ab_testing/wagtail_hooks.py [hunk 1]
```diff
@@ -66,28 +88,41 @@ def register_create_abtest_action_menu_item():
...
-        return ''
+    if WAGTAIL_VERSION >= (2, 15):
+        def render_html(self, context):
+            # v2.15 and later only requires the context object
+            if 'page' in context:
+                return self.format_html(context['request'].user, context)
+
+            return ''
+    else:
+        def render_html(self, request, context):
```
Comment (zerolab) [inline wagtail_ab_testing/wagtail_hooks.py:99]: suggestion: drop support for Wagtail <  2.15 as 2.15 is the LTS and everything before that is long not supported

Comment (nickmoreton) [reply wagtail_ab_testing/wagtail_hooks.py:99]: I like the suggestion. Saw you did this yesterday somewhere 😄

Comment (nickmoreton) [reply wagtail_ab_testing/wagtail_hooks.py]: Support is now for Wagtail >= 2.15

**Review by katdom13 — APPROVED** [review]
These changes look good. Approving.

**Comment by rcoombes-gosh** [comment]
Hi, Is it possible to update the Wagtail version in `install_requires` of `setup.py` to cover v4.1 please?

**Comment by nickmoreton** [comment]
> Hi, Is it possible to update the Wagtail version in `install_requires` of `setup.py` to cover v4.1 please?

Hi Ralph, Thanks for the request. I’m sure that can be done. Any chance you could let us have a pull request for it.
It might be a little while yet before we can get around to it and possibly will be a 4.2 update as well at that point.

**Comment by rcoombes-gosh** [comment]
Hi Nick. Thanks for getting back to me. I'll try and get this done asap. Do you know offhand if this is literally a version number change or should I be looking at other changes made in 4.1 as well?

**Comment by nickmoreton** [comment]
> Hi Nick. Thanks for getting back to me. I'll try and get this done asap. Do you know offhand if this is literally a version number change or should I be looking at other changes made in 4.1 as well?

It's likely it needs other changes for the tests in the tox.ini as well as the setup.py

**Comment by Stormheg** [comment]
superseded by #52

**Pull request closed without merging**